OpenRA.Mods.Common/Activities/Air/ResupplyAircraft.cs

```diff
@@ -48,12 +50,23 @@ public override Activity Tick(Actor self)
 				}
 				else
 				{
-					// Helicopters should take off from their helipad immediately after resupplying.
-					// HACK: Append NextActivity to TakeOff to avoid moving to the Rallypoint (if NextActivity is non-null).
-					inner = ActivityUtils.SequenceActivities(
-						aircraft.GetResupplyActivities(host)
-						.Append(new AllowYieldingReservation(self))
-						.Append(new TakeOff(self)).Append(NextActivity).ToArray());
+                    // Helicopters should take off from their helipad immediately after resupplying,
+                    // unless the "Enter" or "Return to Base" command was explictly given by user
```

Comment:
@GraionDilach this does not affect your production queue to have a helo on the pad. When a helo is in IDLE mode on the pad, new units just pop up from the same pad. This might not be what you expect, but it probably has something to do with the reservation system.

Your build queue is only blocked while a helo is in transit to the pad, or during the resupply animation, but once its idle, they just pop up. I'd be happy to take a video for you, but you should try it.

In TD, the helos will not stay on the pad regardless of what you do, since they are RepairBuilding types. In TS, they will only stay on the pad if they have taken damage, to allow your mechanics to do their work. I have accounted for the use cases to allow the takeoff behavior when desired.